webapp/lib/controller.dart

```diff
@@ -71,6 +75,11 @@ void initUI() {
       needsReplyDataList.removeWhere((d) => updatedIds.contains(d.docId));
       needsReplyDataList.addAll(updatedData);
       command(UIAction.needsReplyDataUpdated, null);
+      
+      if (periodicTimer == null) {
+        checkNeedsReplyMetricsPipelineDataFlow();
+        periodicTimer = Timer.periodic(Duration(hours: 1), (_) => checkNeedsReplyMetricsPipelineDataFlow());
```

Comment:
I don't think you want a period timer here - it will check every hour, meaning that it will not notify us exactly 1h hour after we get some data, but the delay could be up to 2 hours.

What do you think of the following logic:

Have a timer associated with each project
When new data comes in:
    - find out which projects have received new data
    - cancel existing timers for those projects
    - if the updated project is the selected one, mark it as not stale
    - start 1h one-off timers for each of the updated projects, which, when they expire, mark the project as stale (if selected)
When a different project is selected via the dropdown, do the check you're doing now between the latest entry and now: if the difference is longer than 1h, mark the project as stale

Reply:
Makes sense...I see the flaw with my current one. Lemme try out on this logic

Reply:
I've implemented the logic above.